src/View/mainpage/DrawROIWindow.py

```diff
@@ -684,6 +692,46 @@ def set_selected_roi_name(self, roi_name):
         self.roi_name_line_edit.setText(self.ROI_name)
 
 
+class SaveROIProgressWindow(QtWidgets.QDialog):
```

Comment:
this seems to share a fair amount with DeleteROIProgressWindow
Would it make sense to have a base class and derive from it?
It might not be worth it, but once you have started to repeat yourself ("DRY"), it is worth evaluating.